sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -443,6 +453,96 @@ AZ_NODISCARD az_result az_iot_hub_client_methods_response_get_publish_topic(
+ * @brief A command request received from IoT Hub.
+ *
+ */
+typedef struct
+{
+  /**
+   * The request ID.
+   * @note The application must match the command request and command response.
+   */
+  az_span request_id;
+
+  /**
+   * The name of the component which the command was invoked for.
+   * @note Can be `AZ_SPAN_EMPTY` if for the root component.
+   */
+  az_span component_name;
+
+  /**
+   * The command name.
+   */
+  az_span command_name;
+} az_iot_hub_client_command_request;
+
+/**
+ * @brief Attempts to parse a received message's topic for command features.
+ *
+ * @param[in] client The #az_iot_hub_client to use for this call.
+ * @param[in] received_topic An #az_span containing the received topic.
+ * @param[out] out_request If the message is a command request, this will contain the
+ * #az_iot_hub_client_command_request.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p received_topic must be a valid, non-empty #az_span.
+ * @pre \p out_request must not be `NULL`. It must point to an #az_iot_hub_client_command_request
+ * instance.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The topic is meant for this feature and the \p out_request was populated
+ * with relevant information.
+ * @retval #AZ_ERROR_IOT_TOPIC_NO_MATCH The topic does not match the expected format. This could
+ * be due to either a malformed topic OR the message which came in on this topic is not meant for
+ * this feature.
+ */
+AZ_NODISCARD az_result az_iot_hub_client_commands_parse_received_topic(
+    az_iot_hub_client const* client,
+    az_span received_topic,
+    az_iot_hub_client_command_request* out_request);
+
+/**
+ * @brief Gets the MQTT topic that is used to respond to command requests.
+ *
+ * @param[in] client The #az_iot_hub_client to use for this call.
+ * @param[in] request_id The request ID. Must match a received #az_iot_hub_client_command_request
+ * request_id.
+ * @param[in] status A code that indicates the result of the command, as defined by the application.
+ * @param[out] mqtt_topic A buffer with sufficient capacity to hold the MQTT topic. If successful,
+ * contains a null-terminated string with the topic that needs to be passed to the MQTT client.
+ * @param[in] mqtt_topic_size The size, in bytes, of \p mqtt_topic.
+ * @param[out] out_mqtt_topic_length __[nullable]__ Contains the string length, in bytes, of \p
+ * mqtt_topic. Can be `NULL`.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p request_id must be a valid, non-empty #az_span.
+ * @pre \p mqtt_topic must not be `NULL`.
+ * @pre \p mqtt_topic_size must be greater than 0.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The topic was retrieved successfully.
+ */
+AZ_NODISCARD az_result az_iot_hub_client_commands_response_get_publish_topic(
+    az_iot_hub_client const* client,
+    az_span request_id,
+    uint16_t status,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
```

Comment:
>size_t [](http://example.com/codeflow?start=4&length=6)

outside this PR: I was never able to understood why topic length needs more then uint16_t, in spec topic can't exceeds 2 bytes.